benchmark/src/main/java/com/datatorrent/benchmark/monitor/SystemResourceMonitor.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.benchmark.monitor;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+import java.text.NumberFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public class SystemResourceMonitor
+{
+  public static enum SystemResource
+  {
+    /**
+     * memory
+     */
+    CommittedVirtualMemorySize, FreePhysicalMemorySize, TotalPhysicalMemorySize, TotalSwapSpaceSize, FreeSwapSpaceSize,
+
+    /**
+     * cpu
+     */
+    ProcessCpuTime, SystemCpuLoad, ProcessCpuLoad,
+
+    /**
+     * file
+     */
+    OpenFileDescriptorCount, MaxFileDescriptorCount
+  }
+
+  public static SystemResource[] memoryResources = new SystemResource[] {
+      SystemResource.CommittedVirtualMemorySize,
+      SystemResource.FreePhysicalMemorySize,
+      SystemResource.TotalPhysicalMemorySize,
+      SystemResource.TotalSwapSpaceSize,
+      SystemResource.FreeSwapSpaceSize};
+
+  public static SystemResource[] cpuResources = new SystemResource[] {
+      SystemResource.ProcessCpuTime,
+      SystemResource.SystemCpuLoad,
+      SystemResource.ProcessCpuLoad};
+
+  public static SystemResource[] fileResources = new SystemResource[] {
+      SystemResource.OpenFileDescriptorCount,
+      SystemResource.MaxFileDescriptorCount};
+
+  public static SystemResource[] allResources = SystemResource.values();
+
+  private static Map<Set<SystemResource>, SystemResourceMonitor> resourcesToMonitors = Maps.newHashMap();
```

Comment:
My intension is to prevent create two instances monitoring same group of resource